tajo-core/src/main/java/org/apache/tajo/engine/planner/physical/SeqScanExec.java

```diff
@@ -167,23 +162,55 @@ public void init() throws IOException {
       projected = outSchema;
     }
 
-    initScanner(projected);
-    super.init();
-
-    if (plan.hasQual()) {
-      if (scanner.isProjectable()) {
-        qual.bind(context.getEvalContext(), projected);
-      } else {
-        qual.bind(context.getEvalContext(), inSchema);
-      }
+    return projected;
+  }
 
+  private void initScanIterator() {
+    // We should use FilterScanIterator only if underlying storage does not support filter push down.
```

Comment:
We currently have two kinds of scan operator, sequential scan (```SeqScanExec``` and ```PartitionMergeScanExec```) and index scan (```BSTIndexScanExec```). 
In the case of when an index exists for a HDFS table, index scan will do the same operation of ```FullScanIterator```. So, there seems an unified interface for index scan and ```FullScanIterator```.

Maybe, we design a new scanner for HDFS like ```IndexedScanner```. If so, we can also handle index scan in the below code block.
What do you think? Do you have any plans for this?